RecentActivity/src/org/sleuthkit/autopsy/recentactivity/ExtractWebAccountType.java

```diff
@@ -0,0 +1,369 @@
+    }
+    
+    private void extractDomainRoles(Content dataSource, IngestJobContext context) {
+        try {
+            // Get web history blackboard artifacts
+            Collection<BlackboardArtifact> listArtifacts = currentCase.getSleuthkitCase().getBlackboard().getArtifacts(
+                    Arrays.asList(new BlackboardArtifact.Type(BlackboardArtifact.ARTIFACT_TYPE.TSK_WEB_HISTORY)),
+                    Arrays.asList(dataSource.getId()));
+            logger.log(Level.INFO, "Processing {0} blackboard artifacts.", listArtifacts.size()); //NON-NLS
+
+            // Set up collector for roles
+            RoleProcessor roleProcessor = new RoleProcessor(context);
+            
+            // Process each URL
+            for (BlackboardArtifact artifact : listArtifacts) {
+                if (context.dataSourceIngestIsCancelled()) {
+                    return;
+                }
+                
+                findRolesForUrl(artifact, roleProcessor);
+            }
+            
+            // Create artifacts
+            roleProcessor.createArtifacts();
+            
+        } catch (TskCoreException e) {
+            logger.log(Level.SEVERE, "Encountered error retrieving artifacts for domain role analysis", e); //NON-NLS
+        }
+    }
+    
+    /**
+     * Extract and store any role found in the given artifact.
+     * 
+     * @param artifact      The original artifact
+     * @param roleProcessor Object to collect and process domain roles.
+     * 
+     * @throws TskCoreException 
+     */
+    private void findRolesForUrl(BlackboardArtifact artifact, RoleProcessor roleProcessor) throws TskCoreException {
+        
+        BlackboardAttribute urlAttr = artifact.getAttribute(new BlackboardAttribute.Type(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_URL));
+        if (urlAttr == null) {
+            return;
+        }
+
+        BlackboardAttribute domainAttr = artifact.getAttribute(new BlackboardAttribute.Type(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_DOMAIN));
+        if (domainAttr == null) {
+            return;
+        }
+
+        String url = urlAttr.getValueString();
+        String domain = domainAttr.getValueString(); 
+        
+        findMyBbRole(url, domain, artifact, roleProcessor);
+        findPhpBbRole(url, domain, artifact, roleProcessor);
+        findJoomlaRole(url, domain, artifact, roleProcessor);
+        findWordPressRole(url, domain, artifact, roleProcessor);
+    }
+
+    /**
+     * Extract myBB role.
+     * 
+     * @param url       The full URL.
+     * @param domain    The domain.
+     * @param artifact  The original artifact.
+     * @param roleProcessor Object to collect and process domain roles.
+     */    
+    private void findMyBbRole(String url, String domain, BlackboardArtifact artifact, RoleProcessor roleProcessor) {    
+        String platformName = "myBB platform"; // NON-NLS
```

Comment:
Very minor point, but should the platform suffix be a bundle message? Or is that part of the proper noun? Same for all instances of platformName variable in the various methods.

Reply:
I don't think so, since the combination of name + translated "platform" may not make sense in other languages due to word ordering

Reply:
Oh right, good point.